invertedai/common.py

```diff
@@ -2,7 +2,9 @@
 from enum import Enum
 from pydantic import BaseModel, model_validator
 import math
-
+from PIL import Image as PImage
+import numpy as np
```

Comment:
According to the `pyproject.toml` file, `numpy` is not a required dependency, so we shouldn't be importing it here at top level. It's a little concerning that the CI passes despite this.

Reply:
I think it’s because matplotlib is a required dependency and numpy is actually a required package for it.

Reply:
Why is matplotlib required in the first place? Is it just for ScenePlotter? In any case, I think we should make a conscious decision about whether we want to require numpy independently of matplotlib, and if so add it to the dependencies explicitly.